### PR TITLE
fix(ecs): compute_instance support update user_data

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**: compute_instance support update user_data

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.compute_instance support update user_data.
2.user_data support unencrypted value.
3.user_data support base64 encrypted value.
4.user_data, the value will be overwritten during creation, so this value will not be saved.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/ecs" TESTARGS="-run TestAccComputeInstance_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ecs -v -run TestAccComputeInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccComputeInstance_basic
=== PAUSE TestAccComputeInstance_basic
=== CONT  TestAccComputeInstance_basic
--- PASS: TestAccComputeInstance_basic (376.40s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ecs       376.447s
```

### CODE REVIEW

File: docs/resources/compute_instance.md [hunk 1]
```diff
@@ -529,7 +529,7 @@ terraform import huaweicloud_compute_instance.my_instance b11b407c-e604-4e8d-8bc
 
 Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
 API response, security or some other reason.
-The missing attributes include: `admin_pass`, `user_data`, `metadata`, `data_disks`, `scheduler_hints`, `stop_before_destroy`,
+The missing attributes include: `admin_pass`, `metadata`, `data_disks`, `scheduler_hints`, `stop_before_destroy`,
```
Comment (Lance52259) [inline docs/resources/compute_instance.md]: the value of the parameter `user_data` is never save in the Read function, why we remove it?

File: huaweicloud/services/ecs/resource_huaweicloud_compute_instance.go [hunk 2]
```diff
@@ -311,11 +311,10 @@ func ResourceComputeInstance() *schema.Resource {
...
 			"user_data": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				// just stash the hash for state & diff comparisons
-				StateFunc: utils.HashAndHexEncode,
+				Type:             schema.TypeString,
+				Optional:         true,
+				StateFunc:        utils.HashAndHexEncode,
+				DiffSuppressFunc: utils.SuppressUserData,
```
Comment (Lance52259) [inline huaweicloud/services/ecs/resource_huaweicloud_compute_instance.go:319]: please support a comment to tell others what this function does without entry the function.

Comment (wangyuancheng01) [reply huaweicloud/services/ecs/resource_huaweicloud_compute_instance.go:319]: the description of this method has been added.

File: docs/resources/compute_instance.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/resources/compute_instance.md]: I misunderstood, and have already corrected it back.

File: docs/resources/compute_instance.md [hunk 3]
```diff
@@ -289,8 +289,8 @@ The following arguments are supported:
 * `eip_id` - (Optional, String, ForceNew) Specifies the ID of an *existing* EIP assigned to the instance.
   This parameter and `eip_type`, `bandwidth` are alternative. Changing this creates a new instance.
 
-* `user_data` - (Optional, String, ForceNew) Specifies the user data to be injected during the instance creation. Text
-  and text files can be injected. Changing this creates a new instance.
+* `user_data` - (Optional, String) Specifies the user data to be injected during the instance creation. Text
+  and text files can be injected. This value support unencrypted, also support base64 encrypted.
```
Comment (ShiChangkuo) [inline docs/resources/compute_instance.md]:
```suggestion
* `user_data` - (Optional, String) Specifies the user data to be injected to the instance during the creation. Text
  and text files can be injected. The content of `user_data` can be plaint text or encoded with base64.
```

File: huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_instance_test.go [hunk 4]
```diff
@@ -461,6 +466,7 @@ resource "huaweicloud_compute_instance" "test" {
   agency_name         = "test222"
   agent_list          = "ces"
   auto_terminate_time = ""
+  user_data           = ""
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/ecs/resource_huaweicloud_compute_instance_test.go]: this line can be deleted
```suggestion
```

**Pull request merged**